src/pages/Profile.js

```diff
@@ -355,22 +356,131 @@ const PassportLinker = ({ queryParams: { state, code } = {}, provider, prettyNam
 
 const sectionTitle = text => h2({ style: styles.sectionTitle }, [text])
 
-const Profile = ({ queryParams }) => {
-  // State
+const ExternalIdentitiesTab = ({ queryParams }) => {
+  return h(PageBox, { role: 'main', style: { flexGrow: 1 }, variant: PageBoxVariants.LIGHT }, [
+    h(NihLink, { nihToken: queryParams?.['nih-username-token'] }),
+    _.map(provider => h(FenceLink, { key: provider.key, provider }), allProviders),
+    !!getConfig().externalCredsUrlRoot && h(PassportLinker, { queryParams, provider: 'ras', prettyName: 'RAS' })
+  ])
+}
+
+const NotificationCheckbox = ({ notificationKeys, setSaving, prefsData }) => {
+  const notificationKeysWithValue = ({ notificationKeys, value }) => {
+    return Object.assign(...notificationKeys.map(notificationKey => ({ [notificationKey]: `${JSON.stringify(value)}` })))
```

Comment:
Lodash's [fromPairs](https://lodash.com/docs/4.17.15#fromPairs) might be useful here. Something like:
```suggestion
    return _.fromPairs(_.map(notificationKey => [notificationKey, JSON.stringify(value)], notificationKeys))
```